napari/util/misc.py

```diff
@@ -210,3 +211,16 @@ def segment_normal_vector(a, b):
     else:
         unit_norm = normal/norm
     return unit_norm
+
+
+class StringEnum(Enum):
+    def _generate_next_value_(name, start, count, last_values):
+        """ autonaming function assigns each value its own name as a value
+        """
+        return name
```

Comment:
This is so great! But how about using `name.lower()`? Then you can set it with any case input.

```
In [1]: from enum import Enum, auto

In [2]: class LowerName(Enum):
   ...:     def _generate_next_value_(name, start, count, last_values):
   ...:         return name.lower()
   ...:     

In [3]: class Mode(LowerName):
   ...:     PANZOOM = auto()
   ...:     SELECT = auto()
   ...:     

In [4]: x = Mode('panzoom')

In [5]: x
Out[5]: <Mode.PANZOOM: 'panzoom'>
```

Reply:
Sorry, I take that back, then it only works in lowercase...

Reply:
i like lowercase names more because when we're spitting the mode back out to the user, lowercase names look nicer (e.g. `'PICKER'` vs `'picker'`)

Reply:
I totally agree that lowercase looks better to the user. However, the style in [the docs](https://docs.python.org/3/library/enum.html) is to use upper case names (like a constant I guess?). To harmonize these two things, the Enum names are capitol, but in the layer setters, I  call `upper()` on the string (similar to @jni's suggestion, e.g., [`Markers` layer](https://github.com/kevinyamauchi/napari/blob/a7346c9712973420304a72f9d3c4ddbf73d565c8/napari/layers/_markers_layer/model.py#L262)). Thus, our internal representation complies with the style guide and the user can use lower case strings.

I suppose we can also overload the `_missing_` method of `StringEnum` to capitalize strings so `Mode` enums can be instantiated with lower case strings (Python 3.6+ only). For example:
```python
class StringEnum(Enum):
    def _generate_next_value_(name, start, count, last_values):
        """ autonaming function assigns each value its own name as a value
        """
        return name

    def _missing_(cls, value):
        return cls(value.upper())

    def __str__(self):
        """String representation: The string method returns the
        valid vispy symbol string for the Markers visual.
        """
        return self.value.lower()
```

Thoughts?

Reply:
I don't understand – where are you getting the idea that the enum values have to be uppercase too? That was just an example of a `_generate_next_value_` method.

Also big fan of the `_missing_` method over the `.upper()` everywhere.

Reply:
Ah, sorry for the confusion. With regards to upper case names, I meant that the name (as opposed to value) of the Enum member should be upper case. I thought you saying that you preferred lowercase names over upper case names. If you'd prefer the values to be lowercase, I can modify `_generate_next_value_`.

I can implement the _missing_ method and update the setters!

Reply:
I just added a `__missing__` method to `StringEnum` so now the user can instantiate `StringEnum` with any case strings (i.e., `Mode('pan_zoom') == Mode('PAN_ZOOM')`. Also, I updated the `__str__` method such that the Enum string is returned as lower case.

The remaining question is should the `StringEnum` values be upper or lower case strings :P. Currently, they are upper case to match the name. I am happy either way. Any strong preferences @kne42 , @jni, @sofroniewn ?

Reply:
lower